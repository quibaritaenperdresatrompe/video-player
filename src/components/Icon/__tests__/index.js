import Icon from '../';

import itRendersAllMutations from '../../../utils/jest-it-renders-all-mutations';

const mutations = [
  {
    name: 'without props',
    props: null,
  },
  {
    name: 'with a known iconName props',
    props: {
      iconName: 'PlayIcon',
    },
  },
  {
    name: 'with an unknown iconName props',
    props: {
      iconName: 'UnknownIcon',
    },
  },
  {
    name: 'with all props',
    props: {
      color: '#fff',
      hoverColor: '#000',
      iconName: 'PlayIcon',
      size: '3em',
    },
  },
];

describe(Icon.name, () => {
  itRendersAllMutations(Icon, mutations);
});
