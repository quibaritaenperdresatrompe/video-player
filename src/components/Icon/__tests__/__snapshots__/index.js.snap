// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon with a known iconName props it renders correctly 1`] = `
<svg
  className="css-gt3xp3"
  color="hsla(0, 0%, 100%, 0.8)"
  height="1em"
  viewBox="0 0 24 24"
  width="1em"
>
  <path
    d="M8 5v14l11-7z"
  />
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
</svg>
`;

exports[`Icon with all props it renders correctly 1`] = `
<svg
  className="css-5n7asi"
  color="#fff"
  height="3em"
  viewBox="0 0 24 24"
  width="3em"
>
  <path
    d="M8 5v14l11-7z"
  />
  <path
    d="M0 0h24v24H0z"
    fill="none"
  />
</svg>
`;

exports[`Icon with an unknown iconName props it renders correctly 1`] = `null`;

exports[`Icon without props it renders correctly 1`] = `null`;
